File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/HomeScreenInfo.java
package com.pennapps.labs.pennmobile.classes;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by Jackie on 2018-03-28.
 */

public class HomeScreenInfo {

    // news
    @SerializedName("articleUrl")
    @Expose
    private String articleUrl;
    @SerializedName("date")
    @Expose
    private String date;
    @SerializedName("imageUrl")
    @Expose
    private String imageUrl;
    @SerializedName("source")
    @Expose
    private String source;
    @SerializedName("title")
    @Expose
    private String title;


    @SerializedName("description")
    @Expose
    private String description;
    @SerializedName("email")
    @Expose
    private String email;
    @SerializedName("end_time")
    @Expose
    private String endTime;
    @SerializedName("facebook")
    @Expose
    private String facebook;
    @SerializedName("image_url")
    @Expose
    private String imageUrl2;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("start_time")
    @Expose
    private String startTime;
    @SerializedName("website")
    @Expose
    private String website;

    // dining
    @SerializedName("venues")
    @Expose
    private List<Integer> venues;

    // laundry
    @SerializedName("room_id")
    @Expose
    private int roomId;


    public List<Integer> getVenues() {
        return venues;
    }

    public int getRoomId() {
        return roomId;
    }

    public String getArticleUrl() {
        return articleUrl;
    }

    public void setArticleUrl(String articleUrl) {
        this.articleUrl = articleUrl;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getEndTime() {
        return endTime;
    }

    public void setEndTime(String endTime) {
        this.endTime = endTime;
    }

    public String getFacebook() {
        return facebook;
    }

    public void setFacebook(String facebook) {
        this.facebook = facebook;
    }

    public String getImageUrl2() {
        return imageUrl2;
    }

    public void setImageUrl2(String imageUrl2) {
        this.imageUrl2 = imageUrl2;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getStartTime() {
        return startTime;

    }
}

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/HomeScreenInfo.kt
package com.pennapps.labs.pennmobile.classes

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created by Jackie on 2018-03-28. Updated by Marta on 2019-10-20.
 */

class HomeScreenInfo {

    // news
    @SerializedName("article_url")
    @Expose
    val articleUrl: String? = null
    @SerializedName("image_url")
    @Expose
    val imageUrl: String? = null
    @SerializedName("source")
    @Expose
    val source: String? = null
    @SerializedName("title")
    @Expose
    val title: String? = null
    @SerializedName("subtitle")
    @Expose
    val subtitle: String? = null
    @SerializedName("timestamp")
    @Expose
    val timestamp: String? = null

    // dining
    @SerializedName("venues")
    @Expose
    val venues: List<Int>? = null

    // laundry
    @SerializedName("room_id")
    @Expose
    val roomId: Int? = 0

    // courses
    @SerializedName("courses")
    @Expose
    val courses: List<Course>? = null
    @SerializedName("weekday")
    @Expose
    val weekday: String? = null

    // reservations, university events are arrays


}
